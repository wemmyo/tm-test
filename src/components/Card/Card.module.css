.card {
  border: 1px solid gray;
  margin-bottom: 10px;
  padding: 0.9rem;
  width: fit-content;
  border-radius: 5px;
}
.label {
  font-size: 20px;
  font-weight: bold;
}
.card p {
  margin: 0 0 0.5rem 0;
}
