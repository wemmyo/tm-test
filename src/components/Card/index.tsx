import * as React from "react";
import styles from "./Card.module.css";

// interface CardProps {
//   data: {
//     label: string;
//     apr: number;
//     balanceTransferOfferDuration: number;
//     purchaseOfferDuration: number;
//     credit: number;
//   };
// }

// const Card: React.FC<CardProps> = ({ data }) => {
//   const {
//     label,
//     apr,
//     balanceTransferOfferDuration,
//     purchaseOfferDuration,
//     credit,
//   } = data;
//   return (
//     <div className={styles.card}>
//       <p className={styles.label}>{label}</p>
//       <p>
//         Apr: <strong>{(apr * 100).toFixed(1)}%</strong>
//       </p>
//       <p>
//         Balance Transfer Offer Duration:
//         <strong>{balanceTransferOfferDuration} Months</strong>
//       </p>
//       <p>
//         Purchase Offer Duration:
//         <strong>{purchaseOfferDuration} Months</strong>
//       </p>
//       <p>
//         Credit Available: <strong>{credit}</strong>
//       </p>
//     </div>
//   );
// };

// export default Card;

interface CardProps {
  data: {
    label: string;
    apr: number;
    balanceTransferOfferDuration: number;
    purchaseOfferDuration: number;
    credit: number;
  };
}

interface CardState {}

class Card extends React.Component<CardProps, CardState> {
  // state = { :  }
  render() {
    const {
      label,
      apr,
      balanceTransferOfferDuration,
      purchaseOfferDuration,
      credit,
    } = this.props.data;

    return (
      <div className={styles.card}>
        <p className={styles.label}>{label}</p>
        <p>
          Apr: <strong>{(apr * 100).toFixed(1)}%</strong>
        </p>
        <p>
          Balance Transfer Offer Duration:
          <strong>{balanceTransferOfferDuration} Months</strong>
        </p>
        <p>
          Purchase Offer Duration:
          <strong>{purchaseOfferDuration} Months</strong>
        </p>
        <p>
          Credit Available: <strong>{credit}</strong>
        </p>
      </div>
    );
  }
}

export default Card;
