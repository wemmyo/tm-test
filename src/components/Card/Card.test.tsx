import React from "react";
import { shallow } from "enzyme";
import Card from "./index";

const mockData = {
  id: 1,
  label: "Student Life",
  apr: 0.189,
  balanceTransferOfferDuration: 0,
  purchaseOfferDuration: 6,
  credit: 1200,
};

it("renders without crashing", () => {
  expect(shallow(<Card data={mockData} />)).toMatchSnapshot();
});
