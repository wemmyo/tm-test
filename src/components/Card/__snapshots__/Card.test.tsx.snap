// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="card"
>
  <p
    className="label"
  >
    Student Life
  </p>
  <p>
    Apr: 
    <strong>
      18.9
      %
    </strong>
  </p>
  <p>
    Balance Transfer Offer Duration:
    <strong>
      0
       Months
    </strong>
  </p>
  <p>
    Purchase Offer Duration:
    <strong>
      6
       Months
    </strong>
  </p>
  <p>
    Credit Available: 
    <strong>
      1200
    </strong>
  </p>
</div>
`;
